src/pydata_sphinx_theme/theme/pydata_sphinx_theme/components/version-switcher.html

```diff
@@ -3,11 +3,11 @@
 <script>
 document.write(`
   <div class="version-switcher__container dropdown">
-    <button type="button" class="version-switcher__button btn btn-sm navbar-btn dropdown-toggle" data-bs-toggle="dropdown">
+    <button id="versionswitcherbutton" type="button" class="version-switcher__button btn btn-sm navbar-btn dropdown-toggle" data-bs-toggle="dropdown" aria-haspopup="true" aria-controls="versionswitchermenu">
```

Comment:
@trallard here the version switcher button gets an ID and attributes `aria-haspopup` and `aria-controls` (I'm basically copying from the MDN example for the `menu` role)